runtime/permissions/lib.rs

```diff
@@ -509,55 +527,66 @@ impl<T: Descriptor + Hash> UnaryPermission<T> {
     self.query_desc(desc, AllowPartial::TreatAsPartialGranted)
   }
 
-  fn is_granted(&self, desc: Option<&T>) -> bool {
-    Self::list_contains(desc, self.granted_global, &self.granted_list)
+  fn is_granted(&self, desc: Option<&TQuery>) -> bool {
+    match desc {
+      Some(desc) => {
+        self.granted_global || self.granted_list.iter().any(|v| v.matches(desc))
```

Comment:
Why are we using `.matches()` instead of `.stronger_than()` now?

Reply:
There are multiple specific methods. I've grouped them all on query now.